@shogi/app-controller の engine-controller.test.ts が失敗している(CI 未実行で潜在化)

## 概要
PR #82(vitest を CI 実行)の作業中に、`@shogi/app-controller` のユニットテストが
**既存で失敗している**ことが判明した。これまで vitest が CI 実行されていなかったため
潜在化していた退行。

## 失敗内容
```
FAIL packages/app-controller/src/engine-controller.test.ts > createEngineController > dispose で停止と破棄が呼ばれる
AssertionError: expected "vi.fn()" to be called at least once
  ❯ src/engine-controller.test.ts:425:33
```
`dispose` 時に停止/破棄の mock が呼ばれていない。テスト側の期待が実装とずれているか、
`dispose` の実装退行のいずれか。18/19 は pass。

## 対応
- 原因を切り分け(実装退行 or テスト陳腐化)し修正する。
- 修正後、PR #82 の CI test job の allowlist に `@shogi/app-controller` を追加する
  (現状は本失敗のため意図的に除外し、workflow コメントに理由を記載している)。

## 参考
- PR #82(vitest CI 導入、app-controller を除外)

--- a/.github/workflows/js-format-lint.yml
+++ b/.github/workflows/js-format-lint.yml
@@ -152,9 +152,9 @@ jobs:
       # engine-wasm は Wasm ビルド成果物 (pkg/) を参照するため、ビルド後に typecheck を実行
       - run: pnpm turbo typecheck --filter="@shogi/engine-wasm"
       # ユニットテスト (vitest): build 済みの dist / wasm pkg を再利用するため build job 内に配置。
-      # 除外パッケージの内訳と app-controller の既存失敗については #84 参照。
+      # tauri/wasm 前提や test 無しのパッケージは対象外。
       - name: Test (unit)
-        run: pnpm turbo test --filter=web --filter=@shogi/match-client --filter=@shogi/app-core --filter=@shogi/design-system --filter=@shogi/engine-client --filter=@shogi/ui -- --run
+        run: pnpm turbo test --filter=web --filter=@shogi/match-client --filter=@shogi/app-core --filter=@shogi/app-controller --filter=@shogi/design-system --filter=@shogi/engine-client --filter=@shogi/ui -- --run
       - name: sccache stats
         if: always()
         run: sccache --show-stats || true

--- a/packages/app-controller/src/engine-controller.test.ts
+++ b/packages/app-controller/src/engine-controller.test.ts
@@ -393,7 +393,7 @@ describe("createEngineController", () => {
         expect(controller.getState().isAnalyzing).toBe(false);
     });
 
-    it("dispose で停止と破棄が呼ばれる", async () => {
+    it("dispose で破棄が呼ばれ stop は呼ばれない", async () => {
         const mockClient = createMockEngineClient();
         const controller = createEngineController({
             createClient: (_engineId) => mockClient.client,
@@ -422,7 +422,9 @@ describe("createEngineController", () => {
 
         await controller.command.dispose("sente");
 
-        expect(mockClient.stop).toHaveBeenCalled();
+        // dispose は Worker を terminate するため stop() を呼ばない設計
+        // (理由は engine-controller.ts の disposeEngineForSide 実装コメント参照)
+        expect(mockClient.stop).not.toHaveBeenCalled();
         expect(mockClient.dispose).toHaveBeenCalled();
         expect(controller.getState().engineReady.sente).toBe(false);
     });

